### PR TITLE
Bundle test coverage: #46 #47 #48

### PR DESCRIPTION
**Pull request by jasonherald**
## Summary

Bundles three test-coverage items from epic #29 into one PR:

- **#46** — Extracted `relative_time_from_elapsed(Duration) -> String` as a pure helper inside `src/ui/notification_row.rs`. The original `relative_time(SystemTime)` is now a one-line wrapper. Added 7 unit tests in a new `#[cfg(test)]` module: 4 branch tests (now / Nm / Nh / Nd) + 3 boundary tests at 60s, 3600s, 86400s.
- **#47** — Extracted `build_status(unread, dnd) -> WaybarStatus` as a pure helper inside `src/waybar.rs`. `update_status` now composes `build_status` + the disk-write/signal side effects. Added 4 branch tests covering DND, singular pluralization (`unread == 1`), plural (`unread > 1`), and empty.
- **#48** — Added 4 edge-case tests for `clean_markup` in `src/notification.rs` (nested tags, interleaved/malformed nesting, unmatched `<`, unsupported numeric entities) plus a policy docstring on the function recording the choices.

One commit per issue. No CHANGELOG entry — this bundle is internal coverage with zero user-visible impact.

## Test plan

- [x] `make lint` clean locally (fmt + clippy + test + deny + audit).
- [x] Test count: 68 → 83 (+15 new tests across the three issues).
- [x] Manual smoke test against the live compositor (installed via `make upgrade PREFIX=\$HOME/.local BINDIR=\$HOME/.cargo/bin`):
  - [x] `notify-send` still produces a popup.
  - [x] Panel timestamps render ("now" / "Nm" / etc.) — confirms `relative_time` refactor.
  - [x] DND toggle still works — confirms `build_status` extraction.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Expanded unit tests for markup edge cases, relative-time boundaries, and status variations (tests: 68 → 83).

* **Documentation**
  * Added a detailed plan outlining markup-stripping/decoding behavior and a bundled test/PR workflow with pre-PR and smoke-test steps.

* **Refactor**
  * Improved maintainability by extracting pure helpers for relative-time formatting and status payload construction.

### CODE REVIEW

File: docs/superpowers/plans/2026-05-03-bundle-test-coverage-46-47-48.md [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+- Modify: `src/ui/notification_row.rs` — extract pure helper, change wrapper, add `#[cfg(test)]` module.
+
+- [ ] **Step 1: Locate the current `relative_time` implementation**
+
+```bash
+grep -n "fn relative_time\|fn build_row" src/ui/notification_row.rs
+```
+
+Expected: `relative_time` defined around line 105; `build_row` (the call site) defined around line 8.
+
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply docs/superpowers/plans/2026-05-03-bundle-test-coverage-46-47-48.md:71]: @coderabbitai exactly the same anti-pattern we've already learned for this repo. Reworded the "Expected" text to anchor on stable identifiers (`relative_time` near the bottom of the file; `build_row` as the file's primary `pub(crate)` entry point at the top, named via its call to `relative_time(notif.timestamp)`) so the step stays accurate as the file restructures. Fixed in e5c32e1.

**Comment by jasonherald** [comment]
@coderabbitai re. the outside-diff finding on `src/ui/notification_row.rs:109-118` (named constants for relative-time thresholds):

You're right — the extracted helper still hardcoded 60 / 3600 / 86400, which is exactly the magic-number class CLAUDE.md flags ("no magic numbers, named constants"). Same shape as the recent `WAYBAR_REFRESH_SIGNAL_OFFSET` (#33) and `ICON_BELL_*` (#35) treatment.

Applied your suggested patch in 7f186da, with two extensions:

- The constants build on each other (`SECONDS_PER_HOUR = 60 * SECONDS_PER_MINUTE`, `SECONDS_PER_DAY = 24 * SECONDS_PER_HOUR`) so the relationships are self-documenting and a future "should hours run further than 24?" change has one place to look.
- Reworded the test bodies to reference the constants (`Duration::from_secs(SECONDS_PER_MINUTE)` etc.) instead of repeating the literal seconds, so the assertions can't drift if the constants ever change. Test names dropped the bare-numeric suffix (`60s` / `3600s` / `86400s`) in favor of unit-named forms (`under_a_minute` / `under_an_hour` / etc.) for consistency with the new constants.

7 tests still pass; full 83-test suite still green.

File: docs/superpowers/plans/2026-05-03-bundle-test-coverage-46-47-48.md [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+        format!("{}h", secs / 3600)
+    } else {
+        format!("{}d", secs / 86400)
+    }
+}
+
+fn relative_time(timestamp: SystemTime) -> String {
+    relative_time_from_elapsed(timestamp.elapsed().unwrap_or_default())
+}
+```
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply docs/superpowers/plans/2026-05-03-bundle-test-coverage-46-47-48.md:122]: @coderabbitai exactly right — same drift pattern we've hit before. Once 7f186da introduced the named constants in the shipped code, the plan-doc snippets became stale.

Updated both blocks (lines 94-115 and 121-177) in 48b11c6:

- The helper snippet now declares `SECONDS_PER_MINUTE` / `SECONDS_PER_HOUR` / `SECONDS_PER_DAY` as `const u64`s above `relative_time_from_elapsed`, with the relationships built up (`HOUR = 60 * MINUTE`, `DAY = 24 * HOUR`).
- Every `Duration::from_secs(...)` in the test snippet now uses the constants instead of literal seconds, and the test names switched to the unit-named forms (`now_branch_under_a_minute`, `boundary_at_one_minute_transitions_now_to_1m`, etc.).

Two literal `60s, 3600s, 86400s` references intentionally remain — both inside snippets that describe the *originally shipped* commit message and PR body. Changing those would diverge from what actually got published. Same posture taken on the prior bundle's commit-message snippet.
